src/web/vaev-driver/fetcher.h

```diff
@@ -2,18 +2,216 @@
 
 #include <karm-gc/heap.h>
 #include <karm-mime/url.h>
+#include <karm-net/http/http.h>
+#include <karm-sys/chan.h>
+#include <karm-sys/file.h>
 #include <vaev-dom/document.h>
 #include <vaev-style/stylesheet.h>
 
 namespace Vaev::Driver {
 
-Res<Style::StyleSheet> fetchStylesheet(Mime::Url url, Style::Origin origin = Style::Origin::AUTHOR);
+struct ChunkedTransfer : public Io::Writer {
+    bool isDone = false;
+    Io::Writer& _out;
```

Comment:
The fact that you are forcing the 'ChunkedTransfer` seems to have some nasty repercussions on the rest of the code; I noticed at least one use-after-free in the code below.

Reply:
thx, solving this since its related to first comment